cql/evaluate-measure-hapi.sh

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+
+# Usage: ./evaluate-measure.sh -f <query>.cql <server-base>
+
+# Takes a CQL file, creates a Library resource from it, references that from a
+# Measure resource and calls $evaluate-measure on it.
+
+library() {
+cat <<END
+{
+  "resourceType": "Library",
+  "status": "active",
+  "type" : {
+    "coding" : [
+      {
+        "system": "http://terminology.hl7.org/CodeSystem/library-type",
+        "code" : "logic-library"
+      }
+    ]
+  },
+  "content": [
+    {
+      "contentType": "text/cql"
+    }
+  ]
+}
+END
+}
+
+measure() {
+cat <<END
+{
+  "resourceType": "Measure",
+  "status": "active",
+  "subjectCodeableConcept": {
+    "coding": [
+      {
+        "system": "http://hl7.org/fhir/resource-types",
+        "code": "Patient"
+      }
+    ]
+  },
+  "scoring": {
+    "coding": [
+      {
+        "system": "http://terminology.hl7.org/CodeSystem/measure-scoring",
+        "code": "cohort"
+      }
+    ]
+  },
+  "group": [
+    {
+      "population": [
+        {
+          "code": {
+            "coding": [
+              {
+                "system": "http://terminology.hl7.org/CodeSystem/measure-population",
+                "code": "initial-population"
+              }
+            ]
+          },
+          "criteria": {
+            "language": "text/cql",
+            "expression": "InInitialPopulation"
+          }
+        }
+      ]
+    }
+  ]
+}
+END
+}
+
+create-library() {
+  library | jq -cM ".url = \"urn:uuid:$1\" | .name = \"$1\" | .content[0].data = \"$2\""
+}
+
+create-measure() {
+  measure | jq -cM ".url = \"urn:uuid:$1\" | .library[0] = \"Library/$2\" | .subjectCodeableConcept.coding[0].code = \"$3\""
```

Comment:
Does HAPI really not resolve [canonical URL's](https://www.hl7.org/fhir/references.html#canonical)? If so, this would be an issue for HAPI.

Reply:
Currently it does not work to reference a library by the canonical url. I thinks it maybe a bug in the LibraryHelper.class
The reference value "urn:uuid:3b555976-2c83-406f-9a50-2ca1d1993ac3" will execute the else path and call resolveLibraryById

``` java
  public Library resolveLibraryReference(LibraryResolutionProvider<Library> libraryResourceProvider, String reference) {
        if (!reference.startsWith("Library/") && reference.contains("/")) {
            return reference.contains("/Library/") ? (Library)libraryResourceProvider.resolveLibraryByCanonicalUrl(reference) : null;
        } else {
            return (Library)libraryResourceProvider.resolveLibraryById(reference.replace("Library/", ""));
        }
    }
```

Reply:
We need the call to `libraryResourceProvider.resolveLibraryByCanonicalUrl` why should canonical URL's contain the Path `Library`? Can you please open an issue in HAPI or ask in chat.fhir.org before?